admin-dev/themes/new-theme/js/pages/order/create/product-renderer.js

```diff
@@ -47,10 +47,13 @@ export default class ProductRenderer {
     }
 
     const $productsTableRowTemplate = $($(createOrderMap.productsTableRowTemplate).html());
+    const $productsTableGiftRowTemplate = $($(createOrderMap.productsTableGiftRowTemplate).html());
 
     for (const key in products) {
       const product = products[key];
-      const $template = $productsTableRowTemplate.clone();
+
+      const $template = product.gift === true ? $productsTableGiftRowTemplate.clone() : $productsTableRowTemplate.clone();
```

Comment:
```suggestion
      const $template = product.gift === true ? $(createOrderMap.productsTableGiftRowTemplate).clone() : $(createOrderMap.productsTableRowTemplate).clone();
```

Reply:
Why ? Your suggestion is making the line longer 😮 it's harder to read now.
Why clutter this nice statement with meaningful and shortened variable names ?

And why block the PR for such a small thing 😞

Reply:
Because it's a memory optimization : that avoids to recreate a second jquery object for the same element. With this change, we can remove `$productsTableGiftRowTemplate` which become unuseful and memory hungry.

Reply:
<img src="http://www.commitstrip.com/wp-content/uploads/2015/07/Strip-Variables-650-finalenglish1.jpg">
Source: http://www.commitstrip.com/en/2015/07/23/for-a-few-bytes-more/

Reply:
Also: [Make it work, make it right, make it fast](https://tknilsson.com/2018/05/25/programmer-friday-make-it-work-make-it-right-make-it-fast/)

The "make it fast" should happen **at the end**. Right now a lot of code on PrestaShop does not work, and even more is not right.

Moreover, the provided article clearly states it:
_"When you start to optimize for speed, your codebase often takes on more complexity and becomes harder to understand and work with. A highly optimized loop is harder to read. A cache adds complexity and moving parts._

_Many software systems only need to work, they don’t require any special tuning for performance. Consider the usage scenario, the target audience, and do performance testing to figure out if more performance is necessary."_

I think this optimization is decreasing the code readability and that the memory consumption benefit is insignificant. Moreover you are blocking this PR which is a must-have item for 1.7.7 for such an insignificant part of the PR 😞 . Please think about it, I'm asking you to change your mind 🙏  .

Reply:
I think both of you are right, on different perspectives. ;) 

It's true that @zuk3975 's proposal makes it easier to read, it's also true that it increases browser's memory consumption, even though it's probably not that bad (given this is BO, not FO).

A third option, that would both keep @Progi1984 's optimization and @zuk3975 's readability, would be to extract the condition in a properly named function, for example **getProductsTemplate**, since the purpose of this condition is to define products's template (maybe you can think of a better name?)

```js
  const $template = this._getProductsTemplate();
```

And the function:

```js
  _getProductsTemplate() {
    return product.gift === true ? $(createOrderMap.productsTableGiftRowTemplate).clone() : $(createOrderMap.productsTableRowTemplate).clone();
  }
```

But I really think we are nitpicking over tiny details 😃 😃

Reply:
Yep @matthieu-rolland proposal seems the cleaner, the code readability is important but jquery objects are quite heavy and shouldn't be created without reason.

Reply:
The method name might be something like `cloneProductTemplate(product)` though

Reply:
I approve it ;) Thanks @matthieu-rolland